### PR TITLE
Array

### PR DESCRIPTION
**Pull request by qkraudghgh**
- array
    - [x] circular_counter
    - [x] flatten
    - [x] garage
    - [x] longest_non_repeat
    - [x] merge_intervals
    - [x] missing_ranges
    - [x] plus_one
    - [x] rotate_array
    - [x] summary_ranges
    - [x] three_sum
    - [x] two_sum

### CODE REVIEW

**Comment by mingrammer** [comment]
You should use camelCase one as variable and method names

File: array/plus_one_test.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import (
+	"fmt"
+	"testing"
+)
+
+func TestPlusOneFunc(t *testing.T) {
+	t.Log("Test Plus One algorithm")
+	nums := []int{1, 2, 3, 4, 5, 11, 12, 13}
+	fmt.Print("INPUT: ")
+	fmt.Println(nums)
```
Comment (mingrammer) [inline array/plus_one_test.go]: fmt.Println("INPUT: ", nums) is better

File: array/two_sum_test.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+func TestTwoSumFunc(t *testing.T) {
+	t.Log("Test Two Sum algorithm")
+	nums := []int{2, 7, 11, 15}
+	target := 9
+	fmt.Print("INPUT: ")
+	fmt.Print(nums)
+	fmt.Printf(", Target: %v", target)
+	fmt.Println()
```
Comment (mingrammer) [inline array/two_sum_test.go]: `fmt.Printf(", Target: %v\n", target)` is better than using `fmt.Println()` below `fmt.Printf`

**Review by mingrammer — CHANGES_REQUESTED** [review]

**Comment by mingrammer** [comment]
```go
// better
fmt.Println("INPUT: ", ...)
fmt.Println("OUTPUT", ...)

// worse
fmt.Print("INPUT: ")
fmt.Println(...)
fmt.Print("OUTPUT: ")
fmt.Println(...)
```

File: array/circular_counter_test.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestJosepheusFunc(t *testing.T) {
+	t.Log("Test Circular Counter algorithm")
+	temp := []int{1, 2, 3, 4, 5, 6, 7, 8, 9}
+	fmt.Println("INPUT: [1, 2, 3, 4, 5, 6, 7, 8, 9]")
```
Comment (mingrammer) [inline array/circular_counter_test.go]: `fmt.Println("INPUT: ", temp)` is better

File: array/longest_non_repeat_test.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import (
+	"fmt"
+	"testing"
+)
+
+func TestLongestNonRepeatFunc(t *testing.T) {
+	t.Log("Test Longest Non Repeat algorithm")
+	tempString := "abcabcdefbb"
+	output := longestNonRepeat(tempString)
+	fmt.Println("INPUT: 'abcabcdefbb'")
```
Comment (mingrammer) [inline array/longest_non_repeat_test.go]: `fmt.Println("INPUT: ", tempString)` is better

**Review by mingrammer — CHANGES_REQUESTED** [review]

**Comment by qkraudghgh** [comment]
@mingrammer
I write all array algorithm and Fix style
check plz

**Pull request merged**